sdk/common/src/main/java/io/opentelemetry/sdk/entities/EntityProvider.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.entities;
+
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.sdk.resources.Resource;
+
+public interface EntityProvider {
```

Comment:
ResourceProvider?

Reply:
if that's impossible, maybe EntitiesProvider?

Reply:
How about ResourceFromEntitiesProvider so that we can avoid the existing name conflict?